### PR TITLE
Fix issues for nightly build

### PR DESCRIPTION
**Pull request by Kudo**
## Summary

There are two issues from nightly builds.

#### 1. `VERSION_NAME=1000.0.0-main` in  _ReactAndroid/gradle.properties_ 

the solution is to remove unused _ReactAndroid/gradle.properties_  git revert when publishing package.

#### 2. `pod install` error from downloading hermes, e.g. the url is unavailable. `https://github.com/facebook/react-native/releases/download/v0.0.0-20221002-2027-2319f75c8/hermes-runtime-darwin-debug-v0.0.0-20221002-2027-2319f75c8.tar.gz`

fix _hermes-engine.podspec_ to support nightly build and build hermes from main branch.
  
## Changelog

[General] [Fixed] - Fix nightly build issues

## Test Plan

1. i cannot fully test publish-npm.js workflow and it stops at `npm publish`. i can just check at this moment, the _ReactAndroid/gradle.properties_ is right.
2. create a `npx react-native init` project and `yarn add react-native@nightly`. patch `node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec` and try `pod install`

### CODE REVIEW

File: sdks/hermes-engine/hermes-engine.podspec [hunk 1]
```diff
@@ -28,7 +28,7 @@ git = "https://github.com/facebook/hermes.git"
 if ENV.has_key?('HERMES_ENGINE_TARBALL_PATH')
   Pod::UI.puts '[Hermes] Using pre-built Hermes binaries from local path.' if Object.const_defined?("Pod::UI")
   source[:http] = "file://#{ENV['HERMES_ENGINE_TARBALL_PATH']}"
-elsif version == '1000.0.0'
+elsif version == '1000.0.0' || version.start_with?('0.0.0-')
```
Comment (cortinico) [inline sdks/hermes-engine/hermes-engine.podspec:31]: This is going to change soon as we're looking for a place where to store hermes tarballs also for nightlies: https://github.com/react-native-community/discussions-and-proposals/pull/508

**Comment by cortinico** [comment]
cc @kelset as you've been working on this recently

**Comment by kelset** [comment]
hey @Kudo - for context, I've done changes in this area of the code in two places:
* https://github.com/facebook/react-native/pull/34513
* https://github.com/facebook/react-native/pull/34817
lmk if there's anything I need to modify over there to ensure everything works correctly

File: sdks/hermes-engine/hermes-engine.podspec (hunk 1 above)
Comment (Kudo) [reply sdks/hermes-engine/hermes-engine.podspec:31]: that's super helpful where we could test nightly without building hermes from source 🎉

**Comment by Kudo** [comment]
oh no this pr landed first, sorry @kelset 😅 let me know if there's anything i can help for your prs .

**Comment by Kudo** [comment]
thanks @kelset and everyone helps with pr reviews as always.

**Pull request closed without merging**